### DISCUSSION TITLE
Custom BitmapTileSource or BitmapTileSource doesnt work for some urls

### DISCUSSION THREAD

**Discussion by muromir** [mapsforge/vtm#923] [Q&A] [answered]
Hello, I tray to switchover from mapsforge to vtm library in my app. I stuck when using different tileonline providers. For some of them I am getting map on the screen, for other only white screen although using web browser i got the tile e.g 
https://a.tile.opentopomap.org/12/2162/1289.png  I get the answer and using in mapsforge library using
OnlineTileSource onlineTileSource = new OnlineTileSource(new String[]{"a.tile.opentopomap.org"}, 443); also woks fine.

Now using VTM:

UrlTileSource bm = new BitmapTileSource("https://a.tile.opentopomap.org",0,17);
 BitmapTileLayer mBitmapLayer = new BitmapTileLayer(mapView.map(), bm);
                mapView.map().layers().add(mBitmapLayer);
                VectorTileLayer l = mapView.map().setBaseMap(bm);
                mapView.map().layers().add(new LabelLayer(mapView.map(), l));
                theme = mapView.map().setTheme(VtmThemes.OSMARENDER);
or 

UrlTileSource bm =  DefaultSources.OPENSTREETMAP
                        .url("https://a.tile.opentopomap.org")
                        .tileSize(256)
                        .zoomMax(16)
                        .build();
 BitmapTileLayer mBitmapLayer = new BitmapTileLayer(mapView.map(), bm);
                mapView.map().layers().add(mBitmapLayer);
                VectorTileLayer l = mapView.map().setBaseMap(bm);
                mapView.map().layers().add(new LabelLayer(mapView.map(), l));
                theme = mapView.map().setTheme(VtmThemes.OSMARENDER);

I am getting white screen if i replace https://a.tile.opentopomap.org with https://tile.openstreetmap.org all works fine. Do you know what I am doing wrong?

**Comment by devemux86** [comment, marked as answer]
See the [DefaultSources](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/tiling/source/bitmap/DefaultSources.java) for how to create custom tile sources
and the [BitmapTileActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/BitmapTileActivity.java) sample for how to add them on the map.

OpenTopoMap works fine if added properly, as seen in above sample.

**Comment by muromir** [comment]
Thank you @devemux86! I followed the examples and everything works like expected.
